aiohomekit/controller/abstract.py

```diff
@@ -109,11 +112,53 @@ def transport(self) -> Transport:
     def poll_interval(self) -> timedelta:
         """Returns how often the device should be polled."""
 
+    async def _process_disconnected_events(self):
+        pass
+
     def _async_description_update(
         self, description: AbstractDescription | None
     ) -> None:
+        if self.description != description:
+            logger.debug(
+                "%s: Description updated: old=%s new=%s",
+                self.id,
+                self.description,
+                description,
+            )
+
+        repopulate_accessories = False
+        if description and self.description:
+            if description.config_num > self.description.config_num:
```

Comment:
There is a case in the zeroconf transports where self.description starts as none. Those transports don't currently "fire" until we have seen one baseline descripton and then one change.

BLE solves this by making the description mandatory. We could do the same for zeroconf. So far we have not required zeroconf to be functional for "some" level of homekit support though (after pairing, anyway). The change would be relatively simple - just do a `async_find` into `load_pairing` and take the description from that.

Also need to compare against the stored config_num, but I want to decide what to do about ^ first.

Reply:
Considering how much trouble users have with transient MDNS problems (I maybe jaded from maintaining zeroconf), I would avoid making it mandatory and load from disk in this case.  With Ble we have to have the advertisement to connect to the device so we are stuck with the design there.

Reply:
Damn i was hoping i was being jaded and you might give me a compelling reason why it wasn't a problem any more.